### DISCUSSION TITLE
Result div after code block with annotation

### DISCUSSION THREAD

**Discussion by pawamoy** [squidfunk/mkdocs-material#7259] [Q&A]
I'm trying to concatenate a result div after a code block with an annotation, within a tab. The issue is that the annotation is in between the code block and the result div, so the result div is not styled accordingly: it doesn't have borders and it is not concatenated to the block above.

Example:

```md
=== ":simple-pipx: pipx"

    ```bash
    pipx install griffe  # (1)!
    ```

    1. You can also directly run Griffe with `pipx run griffe <args>`. Tool-only installation (no library use).

    <div class="result" markdown>

    [pipx](https://pipx.pypa.io/stable/) allows to install and run Python applications in isolated environments.

    </div>

=== ":simple-pdm: pdm"

    ```bash
    pdm add griffe
    ```

    <div class="result" markdown>

    [PDM](https://pdm-project.org/en/latest/) is an all-in-one solution for Python project and dependency management.

    </div>
```

The PDM tab shows up like this:

![pdmtab](https://github.com/squidfunk/mkdocs-material/assets/3999221/fb04eac4-9573-4e84-9c51-f3c13803379a)

While the pipx tab shows up like this:

![pipxtab](https://github.com/squidfunk/mkdocs-material/assets/3999221/cd8df7ea-bcf6-489f-bbca-bc3188a0ea9c)

I have tried many alternative sequences of markup, but none of them worked.

- Putting the list in the result div breaks the annotation

```md
=== ":simple-pipx: pipx"

    ```bash
    pipx install griffe  # (1)!
    ```

    <div class="result" markdown>

    1. You can also directly run Griffe with `pipx run griffe <args>`. Tool-only installation (no library use).

    [pipx](https://pipx.pypa.io/stable/) allows to install and run Python applications in isolated environments.

    </div>
```

- Wrapping the code block and list in a div just separates the code block from the tab header and doesn't resolve the original issue

```md
=== ":simple-pipx: pipx"

    <div markdown>

    ```bash
    pipx install griffe  # (1)!
    ```

    1. You can also directly run Griffe with `pipx run griffe <args>`. Tool-only installation (no library use).

    </div>

    <div class="result" markdown>

    [pipx](https://pipx.pypa.io/stable/) allows to install and run Python applications in isolated environments.

    </div>
```

- Wrapping everything in the result div has no effect

```md
=== ":simple-pipx: pipx"

    <div class="result" markdown>

    ```bash
    pipx install griffe  # (1)!
    ```

    1. You can also directly run Griffe with `pipx run griffe <args>`. Tool-only installation (no library use).

    [pipx](https://pipx.pypa.io/stable/) allows to install and run Python applications in isolated environments.

    </div>
```

I know the `result` CSS class is not really public API, but can you think of a way to make this work :thinking:?

**Comment by pawamoy** [comment]
I notice that the bullet list ends up being empty after processing (client side I suppose):

```html
<ol hidden="">
<li></li>
</ol>
```

I also notice that removing it restores the expected behavior of the next result div. So maybe additionally/instead to emptying the unordered list and hiding it with `hidden=""`, MfM could remove it?

**Reply by squidfunk** [reply]
Indeed, the result needs to be directly after the code block. The list remains there in case you print a document, which will move annotations back into the list. Otherwise, annotations can not be printed. It's kind of an edge case that's definitely not optimal right now, but I'm not sure how we can improve it without sacrificing other functionality.

**Reply by pawamoy** [reply]
Thanks for the explanation! Probably naive suggestion: isn't it possible to move back the annotations below by re-creating a list? Or maybe you indeed need to keep it there to know if it was an ordered vs unordered list (or to preserve CSS classes on `li`, etc.)?

In any case, I understand that my use-case is quite niche and probably isn't worth lots of effort :smile: I'll find a workaround :slightly_smiling_face: 

Thanks again!

**Reply by squidfunk** [reply]
Yes, that would be possible. As you know, we're currently frying much bigger fish, so it's a low priority, but if someone (you?) wants to go ahead, a PR is more than welcome ☺️ Here's the part that finds and mounts code annotations:

https://github.com/squidfunk/mkdocs-material/blob/ccc8d8491f35a9b37ff2ea4a60503304fab1cf9b/src/templates/assets/javascripts/components/content/code/_/index.ts#L208-L228

And here's where toggling on print is implemented:

https://github.com/squidfunk/mkdocs-material/blob/ccc8d8491f35a9b37ff2ea4a60503304fab1cf9b/src/templates/assets/javascripts/components/content/annotation/list/index.ts#L182-L196

Additionally, please note that we could also've copied the elements and injected them, but this would mean that any event handlers registered by the author for elements inside those annotation list items would be lost and need to be rebound. Thus, I opted for a much more conservative approach: move them out and move them back when necessary.

**Reply by pawamoy** [reply]
Awesome, if PRs are accepted, I might give it a try, thanks for letting me know :slightly_smiling_face: TypeScript is not my usual bowl of soup though :melting_face:

**Reply by squidfunk** [reply]
Sure!
